openmmtools/multistate/sams.py

```diff
@@ -222,17 +234,23 @@ def __init__(self,
         self.log_target_probabilities = log_target_probabilities
         self.state_update_scheme = state_update_scheme
         self.locality = locality
+        self.weight_update = weight_update
         self.update_stages = update_stages
-        self.flatness_criteria = flatness_criteria
-        self.flatness_threshold = flatness_threshold
         self.weight_update_method = weight_update_method
+        self.beta_factor = beta_factor
         self.adapt_target_probabilities = adapt_target_probabilities
         self.gamma0 = gamma0
         self.logZ_guess = logZ_guess
         # Private variables
         # self._replica_neighbors[replica_index] is a list of states that form the neighborhood of ``replica_index``
         self._replica_neighbors = None
         self._cached_state_histogram = None
+        self.criteria = { criteria: value for criteria, value in (('flatness', flatness),  ('round_trips', round_trips),
```

Comment:
I'd do things a little differently here to simplify the code.

1. Save the `minimum_logZ`, `minimum_visits`, etc. independently as individual `_StoredProperty`. You don't really need a validator for these. The old validator was necessary to check the value of the `flatness_criteria` string, but this is no longer necessary as here you're hardcoding the possible criterias in the contructor arguments.
2. I'd implement `self._criteria` as private property so that it gets updated automatically when one of the criterias is changed by the user (i.e. something like this)
```python
@property
def _criteria(self):
    return { criteria: value for criteria, value in (('logZ_flatness', self.logZ_flatness), 
                                                     ('minimum_round_trips', self.minimum_round_trips),
                                                     ('minimum_logZ', self.minimum_logZ), 
                                                     ('minimum_visits', self.minimum_visits))
                                                     if value is not None}
```
This way if a user modifies `sams_sampler.minimum_visits`, then `sams_sampler._criteria` is automatically updated.